lib/ramble/docs/tutorials/running_an_experiment.rst

```diff
@@ -0,0 +1,270 @@
+
+Before you can configure your GROMACS experiments, you'll need to set up a
+workspace. You can call this workspace ``basic_gromacs``.
+
+.. code-block:: console
+
+    $ ramble workspace create basic_gromacs
+
+This will create a workspace for you in:
+
+.. code-block:: console
+
+    $ $ramble_root/var/ramble/workspaces/basic_gromacs
+
+Now you can activate the workspace and view its default configuration.
+
+.. code-block:: console
+
+    $ ramble workspace activate basic_gromacs
+    $ ramble workspace info
+
+You can use the ``ramble workspace info`` command after editing configuration
+files to see how ramble would use the changes you made.
+
+^^^^^^^^^^^^^^^^^^^^^^^^
+Configure the Workspace
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Within the workspace directory, ramble creates a directory named ``configs``.
+This directory contains generated configuration and template files. Each of
+these files can be edited to configure the workspace, and examples will be
+provided below.
+
+The available files are:
+* ``ramble.yaml`` This file describes all aspects of the workspace. This
+includes the software stack, the experiments, and all variables.
+* ``execute_experiment.tpl`` This file is a template shell script that will be
+rendered to execute each of the experiments that ramble generates.
+
+You can edit these files directly or with the command ``ramble workspace edit``.
+
+To begin, you should edit the ``ramble.yaml`` file to set up the configuration
+for your experiments. For this tutorial, replace the default yaml text with the
+contents of ``$ramble_root/examples/basic_gromacs_config.yaml``:
+
+.. code-block:: yaml
+
+    ramble:
+      variables:
+        processes_per_node: 16
+        mpi_command: 'mpirun -n {n_ranks} -ppn {processes_per_node}'
+        batch_submit: '{execute_experiment}'
+      applications:
+        gromacs: # Application name
+          workloads:
+            water_gmx50: # Workload name from application
+              experiments:
+                pme_single_rank: # Arbitrary experiment name
+                  variables:
+                    n_ranks: '1'
+                    n_threads: '1'
+                    size: '0003'
+                    type: 'pme'
+                rf_single_rank:
+                  variables:
+                    n_ranks: '1'
+                    n_threads: '1'
+                    size: '0003'
+                    type: 'rf'
+            water_bare:
+              experiments:
+                pme_single_rank:
+                  variables:
+                    n_ranks: '1'
+                    n_threads: '1'
+                    size: '0003'
+                    type: 'pme'
+                rf_single_rank:
+                  variables:
+                    n_ranks: '1'
+                    n_threads: '1'
+                    size: '0003'
+                    type: 'rf'
+      spack:
+        concretized: true
+        packages:
+          gcc9:
+            spack_spec: gcc@9.3.0 target=x86_64
```

Comment:
We might want to call out that installing compilers that spack doesn't know about could take a while.

And give users a way to define their own compilers if they want to.

Reply:
Is this ok verbiage? "Note that specifying compilers that Spack doesn't have installed may take a while.
To see available compilers, use ``spack compilers``."

Reply:
Yes, I might even suggest linking to: https://spack.readthedocs.io/en/latest/getting_started.html#spack-compilers